appeals/web/src/server/lib/mappers/lpaQuestionnaire.mapper.js

```diff
@@ -0,0 +1,722 @@
+import { convertFromBooleanToYesNo } from '../boolean-formatter.js';
+import { addressToString } from '#lib/address-formatter.js';
+import * as displayFormatter from '#lib/display-page-formatter.js';
+
+// TODO: Limit the input types to constants (not working??)
+/**
+ * @typedef {object} ComponentType
+ * @prop {string} type
+ */
+
+/**
+ * @typedef MappedLPAQInstructions
+ * @type {object}
+ * @prop {LPAQInstructionCollection} lpaQ
+ */
+
+/**
+ * @typedef LPAQInstructionCollection
+ * @type {object}
+ * @prop {Instructions} isListedBuilding
+ * @prop {Instructions} listedBuildingDetails
+ * @prop {Instructions} doesAffectAListedBuilding
+ * @prop {Instructions} affectsListedBuildingDetails
+ * @prop {Instructions} doesAffectAScheduledMonument
+ * @prop {Instructions} inCAOrrelatesToCA
+ * @prop {Instructions} isCorrectAppealType
+ * @prop {Instructions} conservationAreaMap
+ * @prop {Instructions} siteWithinGreenBelt
+ * @prop {Instructions} notifyingParties
+ * @prop {Instructions} lpaNotificationMethods
+ * @prop {Instructions} [siteNotices]
+ * @prop {Instructions} [lettersToNeighbours]
+ * @prop {Instructions} [pressAdvert]
+ * @prop {Instructions} hasRepresentationsFromOtherParties
+ * @prop {Instructions} [representations]
+ * @prop {Instructions} officersReport
+ * @prop {Instructions} siteAccess
+ * @prop {Instructions} isAffectingNeighbouringSites
+ * @prop {Instructions[]} [neighbouringSite]
+ * @prop {Instructions} lpaHealthAndSafety
+ * @prop {Instructions} otherAppeals
+ * @prop {Instructions} newConditions
+ * @prop {Instructions} reviewOutcome
+ */
+
+/**
+ * @typedef Instructions
+ * @type {object}
+ * @prop {DisplayInstructions} display
+ * @prop {InputInstruction[]} [input]
+ * @prop {string} [submitApi]
+ */
+/**
+ * @typedef DisplayInstructions
+ * @type {object}
+ * @prop {SummaryListRowProperties} [summaryListItem]
+ * @prop {StatusTag} [statusTag]
+ * @prop {TableRowProperties[]} [tableItem]
+ */
+/**
+ * @typedef InputInstruction
+ * @type {ComponentType & (RadioProperties | TextInputProperties)}
+ */
+/**
+ * @typedef StatusTag
+ * @type {object}
+ * @prop {string} status
+ * @prop {string} classes
+ */
+
+/**
+ * @param {*} data
+ * @param {string} currentRoute
+ * @returns {MappedLPAQInstructions}
+ */
+
+export function initialiseAndMapLPAQData(data, currentRoute) {
+	/** @type {MappedLPAQInstructions} */
+	let mappedData = {};
+	mappedData.lpaQ = {};
```

Comment:
To save all the jsdoc `@type` comments below, we could mark this as:

```
mappedData.lpaQ = /** @type {Object<string, Instructions|Instructions[]} */{};
```
(I think that's the right type anyway)

(also applies to the appeals one)

Reply:
That doesn't work. It recognises the object but doesn't provide intellisense